helix-core/src/main/java/org/apache/helix/controller/rebalancer/topology/Node.java

```diff
@@ -36,6 +36,11 @@
 
   public Node() { }
 
+  public Node(String name, long id) {
```

Comment:
Shall we just have a complete constructor? This one only has part of the fields initialized.

Reply:
That's the problem of using not final field, I don't know what are must-have fields which are set later. Given we have all set methods for the rest of fields, I'm not making it worse by only selecting interesting fields.

Reply:
I think this is a minor point. Alternatively, you could just use the empty constructor and call set methods but either way I think would work?